main.go

```diff
@@ -92,7 +116,14 @@ func writeOptionsFile(packageName string, node ast.Node) (found bool) {
 			continue
 		}
 
-		if typeSpec.Name.String() != typeName {
+		var typeName string
+		for _, n := range types {
+			if typeSpec.Name.String() == n {
+				typeName = n
+				break
+			}
+		}
+		if typeName == "" {
```

Comment:
i sure do hope the "go contracts" design gets implemented and gives up a standard way to filter